Index: grub2-2.06/util/grub.d/30_os-prober.in
===================================================================
--- grub2-2.06.orig/util/grub.d/30_os-prober.in
+++ grub2-2.06/util/grub.d/30_os-prober.in
@@ -139,6 +139,7 @@ used_osprober_linux_ids=
 wubi=
 
 for OS in ${OSPROBED} ; do
+  OS="`echo ${OS} | sed -e 's/(20)//g'`"
   DEVICE="`echo ${OS} | cut -d ':' -f 1`"
   LONGNAME="`echo ${OS} | cut -d ':' -f 2 | tr '^' ' '`"
   LABEL="`echo ${OS} | cut -d ':' -f 3 | tr '^' ' '`"
