Index: grub2/util/grub.d/30_uefi-firmware.in
===================================================================
--- grub2.orig/util/grub.d/30_uefi-firmware.in	2022-11-02 10:59:36.600245886 +0800
+++ grub2/util/grub.d/30_uefi-firmware.in	2022-11-02 11:06:00.540230142 +0800
@@ -17,6 +17,13 @@
 # You should have received a copy of the GNU General Public License
 # along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
 
+PC_TYPE=$(dmidecode -s system-product-name|awk '{print $NF}')
+case "$PC_TYPE" in
+        KLVV*|KLVU*|PGUV*|PGUW*)
+        hw_support=true
+        ;;
+esac
+
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 datarootdir="@datarootdir@"
@@ -39,9 +46,11 @@
   gettext_printf "System Setup" >/dev/null 2>&1
   gettext_printf "UnionTech OS Restore" >/dev/null 2>&1
 
+if [ "x$hw_support" != x"true" ];then
   cat << EOF
 menuentry '$(gettext_printf "System Setup")' \$menuentry_id_option 'uefi-firmware' {
 	fwsetup
 }
 EOF
 fi
+fi
