From a4257251f4c31d22303c9a1e40c434cb98eb6969 Mon Sep 17 00:00:00 2001
From: zeno <762013373@qq.com>
Date: Mon, 6 Apr 2026 18:37:38 +0000
Subject: [PATCH] Fix partition number calculation when no primary partitions
 exist


Index: a4257251/grub-core/partmap/msdos.c
===================================================================
--- a4257251.orig/grub-core/partmap/msdos.c
+++ a4257251/grub-core/partmap/msdos.c
@@ -196,9 +196,10 @@ grub_partition_msdos_iterate (grub_disk_
 	      if (hook (disk, &p, hook_data))
 		return grub_errno;
 	    }
-	  else if (p.number < 3)
-	    /* If this partition is a logical one, shouldn't increase the
-	       partition number.  */
+	  else if (! grub_msdos_partition_is_empty (e->type)
+	       && p.number < 3)
+	    /* If this partition is a primary one, increase the partition number.
+	       Do not increase for extended partitions or empty slots.  */
 	    p.number++;
 	}
 
