Description: no LoongArch display module
Author: Angie.J <doucaixia@uniontech.com>
Last-Update: 2022-01-20
---
--- grub2-2.04.13.orig/util/grub.d/00_header.in
+++ grub2-2.04.13/util/grub.d/00_header.in
@@ -206,6 +206,7 @@ if [ "x$serial" = x1 ]; then
     echo "${GRUB_SERIAL_COMMAND}"
 fi
 
+if [ "$machine" != "loongarch64" ]; then
 if [ "x$gfxterm" = x1 ]; then
     if [ -n "$GRUB_FONT" ] ; then
        # Make the font accessible
@@ -256,6 +257,7 @@ if [ $ARCH_NAME != "loongson" ] && [ "$m
 EOF
 fi
 fi
+fi
 
 cat <<EOF
 echo
--- grub2-2.04.13.orig/util/grub.d/10_linux.in
+++ grub2-2.04.13/util/grub.d/10_linux.in
@@ -170,7 +170,7 @@ linux_entry ()
   # Use ELILO's generic "efifb" when it's known to be available.
   # FIXME: We need an interface to select vesafb in case efifb can't be used.
   arch=$(uname -m)
-  if [ "$arch" != "mips64" ];then
+  if [ "$arch" != "mips64" ] && [ "$arch" != "loongarch64" ];then
   if [ "x$GRUB_GFXPAYLOAD_LINUX" = x ]; then
       echo "	load_video" | sed "s/^/$submenu_indentation/"
   else
@@ -183,11 +183,11 @@ linux_entry ()
       echo "	gfxmode \$linux_gfx_mode" | sed "s/^/$submenu_indentation/"
   fi
 
-if [ $ARCH_NAME != "loongson" ];then
+if [ $ARCH_NAME != "loongson" ] && [ "$arch" != "loongarch64" ];then
   echo "	insmod gzio" | sed "s/^/$submenu_indentation/"
 fi
   fi
-if [ $ARCH_NAME != "loongson" ];then
+if [ $ARCH_NAME != "loongson" ] && [ "$arch" != "loongarch64" ];then
   echo "	if [ x\$grub_platform = xxen ]; then insmod xzio; insmod lzopio; fi" | sed "s/^/$submenu_indentation/"
 fi
 

